src/lib/snarky_js_bindings/test_module/simple-zkapp.js

```diff
@@ -37,94 +42,207 @@ class SimpleZkapp extends SmartContract {
+
+await fetchAccount({ publicKey: zkappAddress });
+let zkappAccount = Mina.getAccount(zkappAddress);
+
+// we deployed the contract with an initial state of 1
+expectAssertEquals(zkappAccount.appState[0], Field(1));
+
+// the fresh zkapp account shouldn't have any funds
+expectAssertEquals(zkappAccount.balance, UInt64.from(0));
 
-  console.log(data.zkappCommand);
+console.log("deposit funds\n");
+tx = await Mina.transaction(
+  { sender: feePayerAddress, fee: 100_000_000 },
+  () => {
+    zkapp.deposit(UInt64.from(initialBalance));
+  }
+);
+await tx.prove();
+await (await tx.sign([feePayerKey]).send()).wait(waitParams);
+
+await fetchAccount({ publicKey: zkappAddress });
+zkappAccount = Mina.getAccount(zkappAddress);
+
+// we deposit 10_000_000_000 funds into the zkapp account
+expectAssertEquals(zkappAccount.balance, UInt64.from(initialBalance));
+
+console.log("update 1\n");
+tx = await Mina.transaction(
+  { sender: feePayerAddress, fee: 100_000_000 },
+  () => {
+    zkapp.update(Field(30));
+  }
+);
+await tx.prove();
+await (await tx.sign([feePayerKey]).send()).wait(waitParams);
+
+await fetchAccount({ publicKey: zkappAddress });
+zkappAccount = Mina.getAccount(zkappAddress);
+
+// no balance change expected
+expectAssertEquals(zkappAccount.balance, UInt64.from(initialBalance));
+
+// we updated the zkapp state to 31. x = x.add(y) ---- 31 = 1 + 30
+expectAssertEquals(zkappAccount.appState[0], Field(31));
+
+console.log("update 2\n");
+tx = await Mina.transaction(
+  { sender: feePayerAddress, fee: 100_000_000 },
+  () => {
+    zkapp.update(Field(100));
+  }
+);
+await tx.prove();
+await (await tx.sign([feePayerKey]).send()).wait(waitParams);
+
+await fetchAccount({ publicKey: zkappAddress });
+zkappAccount = Mina.getAccount(zkappAddress);
+
+// no balance change expected
+expectAssertEquals(zkappAccount.balance, UInt64.from(initialBalance));
+
+// we updated the zkapp state to 131
+expectAssertEquals(zkappAccount.appState[0], Field(131));
+
+console.log("payout 1\n");
+tx = await Mina.transaction(
+  { sender: feePayerAddress, fee: 100_000_000 },
+  () => {
+    AccountUpdate.fundNewAccount(feePayerAddress);
+    zkapp.payout(privilegedKey);
+  }
+);
+await tx.prove();
+await (await tx.sign([feePayerKey]).send()).wait(waitParams);
+
+await fetchAccount({ publicKey: zkappAddress });
+zkappAccount = Mina.getAccount(zkappAddress);
+
+// we withdraw (payout) half of the initial balance
+expectAssertEquals(zkappAccount.balance, UInt64.from(initialBalance / 2));
+
+console.log("payout 2 (expected to fail)\n");
+tx = await Mina.transaction(
+  { sender: feePayerAddress, fee: 100_000_000 },
+  () => {
+    AccountUpdate.fundNewAccount(feePayerAddress);
+    zkapp.payout(privilegedKey);
+  }
+);
+
+// this tx should fail, but we wont know that here - so we just check that no state has changed
+await tx.prove();
+await (await tx.sign([feePayerKey]).send()).wait(waitParams);
```

Comment:
actually, there's `TransactionId.isSuccess` that you could check to assert that the tx failed:
```ts
let txId = await tx.sign([feePayerKey]).send();
if (txId.isSuccess) throw Error('expected tx to fail');
```

Reply:
Ohh nice didn't know we had that!

Reply:
Unfortunately that didn't work :( the tx is accepted by the network but won't be successfully applied (invalid preconditions), ye `isSuccess` is flagged true

Reply:
lol we should fix isSuccess :D but merge this without it 👍🏻

Reply:
I think it might not correctly parse the error format that's returned by the node